"buf check breaking" panic with config.IgnoreUnstablePackages and deleted file

I have a package I'm renaming to a "unstable" version number, and am getting segfaults from `buf check breaking` as the ignore func gets a nil `descriptor`:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x143246b]

goroutine 291 [running]:
github.com/bufbuild/buf/internal/buf/bufcheck/internal.(*Runner).newIgnoreFunc.func1(0x188db0a, 0xe, 0x0, 0x0, 0x0, 0x0, 0x176eb80)
	github.com/bufbuild/buf/internal/buf/bufcheck/internal/runner.go:96 +0xab
github.com/bufbuild/buf/internal/buf/bufcheck/internal.(*Helper).AddFileAnnotationf(0xc000460cc0, 0x0, 0x0, 0x0, 0x0, 0x18a042c, 0x27, 0xc00089c020, 0x1, 0x1)
	github.com/bufbuild/buf/internal/buf/bufcheck/internal/helper.go:49 +0x1f1
github.com/bufbuild/buf/internal/buf/bufcheck/bufbreaking/internal/bufbreakingcheck.checkFileNoDelete(0xc00089c010, 0xc0000ce420, 0x15, 0x15, 0xc0000ce6e0, 0x15, 0x15, 0x0, 0x0)
	github.com/bufbuild/buf/internal/buf/bufcheck/bufbreaking/internal/bufbreakingcheck/bufbreakingcheck.go:377 +0x20d
github.com/bufbuild/buf/internal/buf/bufcheck/bufbreaking/internal/bufbreakingcheck.newFilesCheckFunc.func1(0x188db0a, 0xe, 0xc000845ae0, 0xc0000ce420, 0x15, 0x15, 0xc0000ce6e0, 0x15, 0x15, 0x1, ...)
	github.com/bufbuild/buf/internal/buf/bufcheck/bufbreaking/internal/bufbreakingcheck/util.go:36 +0xfd
github.com/bufbuild/buf/internal/buf/bufcheck/internal.(*Checker).check(...)
	github.com/bufbuild/buf/internal/buf/bufcheck/internal/checker.go:86
github.com/bufbuild/buf/internal/buf/bufcheck/internal.(*Runner).Check.func1(0xc00011f6c0, 0xc000845ae0, 0xc0000ce420, 0x15, 0x15, 0xc0000ce6e0, 0x15, 0x15, 0xc000f8c660)
	github.com/bufbuild/buf/internal/buf/bufcheck/internal/runner.go:66 +0xab
created by github.com/bufbuild/buf/internal/buf/bufcheck/internal.(*Runner).Check
	github.com/bufbuild/buf/internal/buf/bufcheck/internal/runner.go:65 +0x2f6
```

Adding a nil check and returning true (or false, if we want to fail in this case) seems to do the trick.

--- a/internal/buf/bufcheck/bufbreaking/bufbreaking_test.go
+++ b/internal/buf/bufcheck/bufbreaking/bufbreaking_test.go
@@ -309,6 +309,15 @@ func TestRunBreakingFileNoDelete(t *testing.T) {
 	)
 }
 
+func TestRunBreakingFileNoDeleteUnstable(t *testing.T) {
+	// https://github.com/bufbuild/buf/issues/211
+	testBreaking(
+		t,
+		"breaking_file_no_delete_unstable",
+		bufanalysistesting.NewFileAnnotationNoLocationOrPath(t, "FILE_NO_DELETE"),
+	)
+}
+
 func TestRunBreakingFileSamePackage(t *testing.T) {
 	testBreaking(
 		t,

--- a/internal/buf/bufcheck/bufbreaking/testdata/breaking_file_no_delete_unstable/a/v1beta1/a.proto
+++ b/internal/buf/bufcheck/bufbreaking/testdata/breaking_file_no_delete_unstable/a/v1beta1/a.proto
@@ -1,0 +1,3 @@
+syntax = "proto3";
+
+package a.v1beta1;

--- a/internal/buf/bufcheck/bufbreaking/testdata/breaking_file_no_delete_unstable/buf.yaml
+++ b/internal/buf/bufcheck/bufbreaking/testdata/breaking_file_no_delete_unstable/buf.yaml
@@ -1,0 +1,3 @@
+version: v1beta1
+breaking:
+  ignore_unstable_packages: true

--- a/internal/buf/bufcheck/bufbreaking/testdata_previous/breaking_file_no_delete_unstable/a/v1beta1/a.proto
+++ b/internal/buf/bufcheck/bufbreaking/testdata_previous/breaking_file_no_delete_unstable/a/v1beta1/a.proto
@@ -1,0 +1,3 @@
+syntax = "proto3";
+
+package a.v1beta1;

--- a/internal/buf/bufcheck/bufbreaking/testdata_previous/breaking_file_no_delete_unstable/a/v1beta1/b.proto
+++ b/internal/buf/bufcheck/bufbreaking/testdata_previous/breaking_file_no_delete_unstable/a/v1beta1/b.proto
@@ -1,0 +1,3 @@
+syntax = "proto3";
+
+package a.v1beta1;

--- a/internal/buf/bufcheck/internal/runner.go
+++ b/internal/buf/bufcheck/internal/runner.go
@@ -93,6 +93,9 @@ func (r *Runner) newIgnoreFunc(config *Config) IgnoreFunc {
 			return true
 		}
 		if config.IgnoreUnstablePackages {
+			if descriptor == nil {
+				return false
+			}
 			packageVersion, ok := protoversion.NewPackageVersionForPackage(descriptor.File().Package())
 			if !ok {
 				return false

